File: codewars/5 kyu Primes in numbers.ts
export class G964 {

  public static primeFactors = (n:number) => {
      var i = 2, count, output = '';
      while(n > 1) {
        count = 0;
        while(n % i == 0) {
          count++;
          n /= i;
        }
        switch(count) {
          case 0 : output += ""; break;
          case 1 : output += "(" +i + ")"; break;
          default : output += "(" +i + "**" +count + ")";
        }
        i++;
      }
    return output;
  }
}

File: codewars/5_kyu_Primes_in_numbers.py


def primeFactors(n):
    i = 2
    count = None
    output = ''
    while n > 1:
      count = 0
      while n % i == 0:
        count += 1
        n /= i
      if count == 0:
        pass
      elif count == 1:
        output += '({})'.format(i)
      else:
        output += '({}**{})'.format(i, count)
      i += 1
    return output

from TestCodewar import Test
test = Test()
test.assert_equals(primeFactors(7775460), "(2**2)(3**3)(5)(7)(11**2)(17)")